Sources/Commands/PackageCommands/DumpCommands.swift

```diff
@@ -52,54 +52,76 @@ struct DumpSymbolGraph: AsyncSwiftCommand {
         //
         // We turn build manifest caching off because we need the build plan.
         let buildSystem = try await swiftCommandState.createBuildSystem(
-            explicitBuildSystem: .native,
             // We are enabling all traits for dumping the symbol graph.
             traitConfiguration: .init(enableAllTraits: true),
             cacheBuildManifest: false
         )
-        try await buildSystem.build()
+        let buildResult = try await buildSystem.build(subset: .allExcludingTests, buildOutputs: [.symbolGraph, .buildPlan])
 
-        // Configure the symbol graph extractor.
-        let symbolGraphExtractor = try SymbolGraphExtract(
-            fileSystem: swiftCommandState.fileSystem,
-            tool: swiftCommandState.getTargetToolchain().getSymbolGraphExtract(),
-            observabilityScope: swiftCommandState.observabilityScope,
-            skipSynthesizedMembers: skipSynthesizedMembers,
-            minimumAccessLevel: minimumAccessLevel,
-            skipInheritedDocs: skipInheritedDocs,
-            includeSPISymbols: includeSPISymbols,
-            emitExtensionBlockSymbols: extensionBlockSymbolBehavior == .emitExtensionBlockSymbols,
-            outputFormat: .json(pretty: prettyPrint)
-        )
+        let symbolGraphDirectory = try swiftCommandState.productsBuildParameters.dataPath.appending("symbolgraph")
 
-        // Run the tool once for every library and executable target in the root package.
-        let buildPlan = try buildSystem.buildPlan
-        let modulesGraph = try await buildSystem.getPackageGraph()
-        let symbolGraphDirectory = buildPlan.destinationBuildParameters.dataPath.appending("symbolgraph")
-        for description in buildPlan.buildModules {
-            guard description.module.type == .library,
-                  modulesGraph.rootPackages[description.package.id] != nil
-            else {
-                continue
-            }
+        let fs = swiftCommandState.fileSystem
+
+        try? fs.removeFileTree(symbolGraphDirectory)
+        try fs.createDirectory(symbolGraphDirectory, recursive: true)
 
-            print("-- Emitting symbol graph for", description.module.name)
-            let result = try symbolGraphExtractor.extractSymbolGraph(
-                for: description,
-                outputRedirection: .collect(redirectStderr: true),
-                outputDirectory: symbolGraphDirectory,
-                verboseOutput: swiftCommandState.logLevel <= .info
+        if buildResult.symbolGraph {
```

Comment:
It might be nice if the BuildResult could include the symbol graph paths themselves so we can compute those at the build system layer instead of in the dump command